### PR TITLE
Add datasource module

### PR DESCRIPTION
**Pull request by cx1111**
Adding a new module to allow users to configure multiple explicit data sources.

Addresses #367

### CODE REVIEW

**Comment by cx1111** [comment]
These interfaces are very liable to change. We should finalize the initial version after adding at least one cloud data source.

Also, please give suggestions on associating credentialing information to these DataSources.

File: wfdb/io/datasource.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+
+_PHYSIONET_DATA_SOURCE = DataSource(
+    "physionet",
+    DataSourceType.HTTP,
+    "http://physionet.org/content/",
+)
+
+# Dict of configured data sources
+_data_sources: Dict[str, DataSource] = {"physionet": _PHYSIONET_DATA_SOURCE}
```
Comment (tompollard) [inline wfdb/io/datasource.py:49]: Should _data_sources be all caps? Feels slightly odd to have this variable hanging around here.

File: wfdb/io/datasource.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                raise ValueError(
+                    "uri field for an HTTP DataSource must be a valid URL"
+                )
+        self._uri = value
+
+
+_PHYSIONET_DATA_SOURCE = DataSource(
+    "physionet",
+    DataSourceType.HTTP,
+    "http://physionet.org/content/",
```
Comment (tompollard) [inline wfdb/io/datasource.py:45]: https?

**Review by tompollard — APPROVED** [review]
Looks good to me! Added a couple of minor comments, and it looks like there is a conflict that needs dealing with before we merge.

File: wfdb/io/datasource.py (hunk 1 above)
Comment (cx1111) [reply wfdb/io/datasource.py:49]: It's a global (module level) variable for storing the data sources. Not a constant.

File: wfdb/io/datasource.py (hunk 2 above)
Comment (cx1111) [reply wfdb/io/datasource.py]: Done.

**Comment by cx1111** [comment]
> Why does this point to http://physionet.org/content/ and not https://physionet.org/files/?

Fixed

**Pull request merged**